### ISSUE TITLE
Understanding loss and Vrd_Model

### ISSUE THREAD

**Issue by vthost** [GriffinLiang/vrd-dsr#35]
Hi,

- The loss function you give in the paper differs from what the code does (it seems to me). I see that nn.MultiLabelMarginLoss is one part, but I do not understand lines 24 and 29 in lib/model.py. Could you maybe explain that?
- Similarly, I do not see where the paper describes Vrd_Model. It says "...we choose VGG16 (Simonyan and Zisserman 2014) as the base network and the RoI features are further fed into two fully connected layers". But you have extra convolutional layers etc. Could you maybe explain the reason for them or point me to a part I am missing in the paper?
- In every training epoch, you extract all the features in VrdDataLayer.forward(). Do they change with different epochs?

I am rather new in the field, so some things might be obvious... Sorry if this is the case.
Thank you so much for providing the code!
